project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentContent.tsx

```diff
@@ -120,58 +95,122 @@ export const TransportAndPaymentContent: FC<TransportAndPaymentContentProps> = (
-            return getGtmPickupPlaceFromStore(lastOrder.pickupPlaceIdentifier, pickupPlaceData.store);
-        }
-
-        return getGtmPickupPlaceFromLastOrder(lastOrder.pickupPlaceIdentifier, lastOrder);
-    }, [lastOrder, packeteryPickupPoint, pickupPlaceData?.store]);
+    useEffect(() => {
+        setIsLoadingInitialTransportAndPaymentData((!lastOrderData && isUserLoggedIn) || !transportsData);
+    }, [lastOrderData, isUserLoggedIn, transportsData]);
 
     return (
         <>
-            {transports !== undefined && transports.length > 0 && (
-                <TransportAndPaymentSelect
-                    transports={transports}
-                    lastOrderPickupPlace={lastOrderPickupPlace}
-                    lastOrderTransportUuid={lastOrder?.transport.uuid ?? null}
-                    lastOrderPaymentUuid={lastOrder?.payment.uuid ?? null}
-                    changeTransportInCart={changeTransportInCart}
-                    changePaymentInCart={changePaymentInCart}
-                    isTransportSelectionLoading={isTransportSelectionLoading}
-                />
-            )}
-            <OrderAction
-                buttonBack={t('Back')}
-                buttonNext={t('Contact information')}
-                hasDisabledLook={
-                    hasValidationErrors(transportAndPaymentValidationMessages) ||
-                    isTransportSelectionLoading ||
-                    isPaymentSelectionLoading
-                }
-                isLoading={
-                    (isTransportSelectionLoading || isPaymentSelectionLoading) && transport !== null && payment !== null
-                }
-                withGapTop
-                withGapBottom
-                buttonBackLink={cartUrl}
-                nextStepClickHandler={onSelectTransportAndPaymentHandler}
-            />
-            {isErrorPopupVisible && (
-                <ErrorPopup
-                    onCloseCallback={() => setErrorPopupVisibility(false)}
-                    fields={transportAndPaymentValidationMessages}
-                    gtmMessageOrigin={GtmMessageOriginType.transport_and_payment_page}
-                />
-            )}
+            <OrderLayout
+                activeStep={2}
+                isTransportOrPaymentLoading={Boolean(isTransportSelectionLoading) || isPaymentSelectionLoading}
+                isLoading={isTransportAndPaymentSkeletonVisible}
+            >
+                {isTransportAndPaymentSkeletonVisible ? (
+                    <Skeleton className="h-64 w-full" />
+                ) : (
+                    <>
+                        {!!transportsData?.transports.length && (
+                            <TransportAndPaymentSelect
+                                transports={transportsData.transports}
+                                lastOrderTransportUuid={lastOrderData?.lastOrder?.transport.uuid ?? null}
+                                lastOrderPaymentUuid={lastOrderData?.lastOrder?.payment.uuid ?? null}
```

Comment:
Can we work only with `undefined` or defined value? 

 ```suggestion
                                lastOrderTransportUuid={lastOrderData?.lastOrder?.transport.uuid}
                                lastOrderPaymentUuid={lastOrderData?.lastOrder?.payment.uuid}
```

Reply:
After doing the change (and also required subsequent changes), I dropped it and decided not to do it right now. I think we need to handle this more generally.